core/src/main/java/org/infinispan/persistence/sifs/Index.java

```diff
@@ -147,11 +149,50 @@ static byte[] toIndexKey(int cacheSegment, byte[] bytes, int offset, int length)
    /**
     * @return True if the index was loaded from well persisted state
     */
-   public boolean isLoaded() {
+   public boolean load() {
+      if (!checkForExistingIndexSizeFile()) {
+         return false;
+      }
+      try {
+         File statsFile = new File(indexDir.toFile(), "index.stats");
+         if (!statsFile.exists()) {
+            return false;
+         }
+         FileChannel statsChannel = new RandomAccessFile(statsFile, "rw").getChannel();
+
+         // id / length / free / expirationTime
```

Comment:
you probably want to add a "magic" byte in case we need to change the format later.

Reply:
Sure, we can do that.